sdlib/d/gc/arena.d

```diff
@@ -240,6 +269,28 @@ private:
 		(cast(Arena*) &this).freePagesImpl(e, n, pages);
 	}
 
+	void shrinkAlloc(shared(ExtentMap)* emap, Extent* e, uint delta) shared {
+		assert(!e.isHuge(), "Does not support huge!");
+		assert(isAligned(e.address, PageSize), "Invalid extent address!");
+		assert(isAligned(e.size, PageSize), "Invalid extent size!");
+		assert(delta > 0, "No change in number of pages!");
+
+		assert(e.hpd.address is alignDown(e.address, HugePageSize),
+		       "Invalid hpd!");
+
+		uint n = ((cast(size_t) e.address) / PageSize) % PageCount;
+
+		uint oldPages = cast(uint) e.size / PageSize;
```

Comment:
For instance here (in reference to https://github.com/snazzy-d/sdc/pull/289/files#r1318633761 ).